<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>650</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Image Encryptor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/windowIcon</normaloff>:/windowIcon</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout_7">
      <item row="3" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="buttonSetRandomParameters">
          <property name="text">
           <string>Random Parameters</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="buttotnEncrypt">
          <property name="text">
           <string>Encrypt</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonDecrypt">
          <property name="text">
           <string>Decrypt</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QTabWidget" name="tabWidgetImage">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Source Image</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QScrollArea" name="scrollArea">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>506</width>
               <height>367</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="srcImageLabel">
                <property name="text">
                 <string/>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Ð¡ipher</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QScrollArea" name="scrollArea_2">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents_2">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>506</width>
               <height>367</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="dstImageLabel">
                <property name="text">
                 <string/>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="1" column="0" alignment="Qt::AlignBottom">
       <widget class="QTabWidget" name="tabWidgetEncryptionMethod">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="tabBarAutoHide">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>DNA Algorithm</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <property name="bottomMargin">
             <number>5</number>
            </property>
            <item row="0" column="0" colspan="4" alignment="Qt::AlignBottom">
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Initial Conditions</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QLineEdit" name="lineEditDNAy0">
              <property name="minimumSize">
               <size>
                <width>90</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_x0">
              <property name="text">
               <string>x0:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="lineEditDNAx0">
              <property name="minimumSize">
               <size>
                <width>90</width>
                <height>0</height>
               </size>
              </property>
              <property name="inputMask">
               <string/>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_y0">
              <property name="text">
               <string>y0:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Hash:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="5">
             <widget class="QLineEdit" name="lineEditDNAz0">
              <property name="minimumSize">
               <size>
                <width>90</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item row="1" column="6">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Chaotic Map:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="7">
             <widget class="QComboBox" name="comboBoxDNAEncoding">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>1</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="currentText">
               <string>1</string>
              </property>
              <item>
               <property name="text">
                <string>1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>3</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>4</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>6</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>7</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>8</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="4">
             <widget class="QLabel" name="label_z0">
              <property name="text">
               <string>z0:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="7">
             <widget class="QComboBox" name="comboBoxDNAChaoticMap">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <item>
               <property name="text">
                <string>Lorenz</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>PWLCM</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Logistic</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Tent</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Baker</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="2" column="1" colspan="5">
             <widget class="QLineEdit" name="lineEditDNAhash">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>10</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="inputMask">
               <string/>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="maxLength">
               <number>64</number>
              </property>
             </widget>
            </item>
            <item row="2" column="6">
             <widget class="QLabel" name="label_7">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>5</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Encoding:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_4">
         <attribute name="title">
          <string>Yoon Algorithm</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="1" column="2">
           <widget class="QLineEdit" name="lineEditYoonX"/>
          </item>
          <item row="1" column="5">
           <widget class="QLineEdit" name="lineEditYoonP"/>
          </item>
          <item row="1" column="7">
           <widget class="QLabel" name="label_9">
            <property name="minimumSize">
             <size>
              <width>45</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>K</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="label_8">
            <property name="minimumSize">
             <size>
              <width>45</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>P</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_6">
            <property name="minimumSize">
             <size>
              <width>45</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>X</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="8" colspan="2">
           <widget class="QLineEdit" name="lineEditYoonK"/>
          </item>
          <item row="0" column="0" colspan="10" alignment="Qt::AlignBottom">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Parameters</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_5">
         <attribute name="title">
          <string>EA</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="1" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Population Size</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="EApopulationSpinBox">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>2</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Mutation Rate</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QDoubleSpinBox" name="EAmutationRateSpinBox">
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QDoubleSpinBox" name="EAenthropySpinBox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>8.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Enthropy</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="5">
           <widget class="QLineEdit" name="lineEdit_EADeckey"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>Decryption Key</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>550</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEncryption">
    <property name="title">
     <string>Encryption</string>
    </property>
    <addaction name="actionEncrypt"/>
    <addaction name="actionDecrypt"/>
   </widget>
   <widget class="QMenu" name="menuAnalisys">
    <property name="title">
     <string>Analysis</string>
    </property>
    <widget class="QMenu" name="menuStatistical_Analisys">
     <property name="title">
      <string>Statistical Analisys</string>
     </property>
     <widget class="QMenu" name="menuCorrelation">
      <property name="title">
       <string>Correlation</string>
      </property>
      <addaction name="actionHorizontal"/>
      <addaction name="actionVertical"/>
      <addaction name="actionDiagonal"/>
     </widget>
     <addaction name="actionHistogram"/>
     <addaction name="menuCorrelation"/>
     <addaction name="actionEnthropy"/>
    </widget>
    <widget class="QMenu" name="menuDifferential_Analisys">
     <property name="title">
      <string>Differential Analisys</string>
     </property>
     <addaction name="actionNPCR"/>
     <addaction name="actionUACI"/>
    </widget>
    <addaction name="menuStatistical_Analisys"/>
    <addaction name="menuDifferential_Analisys"/>
    <addaction name="actionEncryption_time"/>
    <addaction name="actionFull_Analysis"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionPlain_Cipher"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_Image_Encryptor"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuEncryption"/>
   <addaction name="menuAnalisys"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionEncrypt">
   <property name="text">
    <string>Encrypt</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionDecrypt">
   <property name="text">
    <string>Decrypt</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionPlain_Cipher">
   <property name="text">
    <string>Plain&lt;-&gt;Cipher</string>
   </property>
  </action>
  <action name="actionHistogram">
   <property name="text">
    <string>Histogram</string>
   </property>
  </action>
  <action name="actionHorizontal">
   <property name="text">
    <string>Horizontal</string>
   </property>
  </action>
  <action name="actionVertical">
   <property name="text">
    <string>Vertical</string>
   </property>
  </action>
  <action name="actionEnthropy">
   <property name="text">
    <string>Enthropy</string>
   </property>
  </action>
  <action name="actionDiagonal">
   <property name="text">
    <string>Diagonal</string>
   </property>
  </action>
  <action name="actionNPCR">
   <property name="text">
    <string>NPCR</string>
   </property>
  </action>
  <action name="actionUACI">
   <property name="text">
    <string>UACI</string>
   </property>
  </action>
  <action name="actionFull_Analysis">
   <property name="text">
    <string>Full Analysis</string>
   </property>
   <property name="toolTip">
    <string>Full Analysis</string>
   </property>
  </action>
  <action name="actionEncryption_time">
   <property name="text">
    <string>Encryption time</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionAbout_Image_Encryptor">
   <property name="text">
    <string>About Image Encryptor</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>tabWidgetEncryptionMethod</tabstop>
  <tabstop>lineEditDNAx0</tabstop>
  <tabstop>lineEditDNAy0</tabstop>
  <tabstop>lineEditDNAz0</tabstop>
  <tabstop>comboBoxDNAChaoticMap</tabstop>
  <tabstop>comboBoxDNAEncoding</tabstop>
  <tabstop>lineEditYoonX</tabstop>
  <tabstop>lineEditYoonP</tabstop>
  <tabstop>lineEditYoonK</tabstop>
  <tabstop>buttotnEncrypt</tabstop>
  <tabstop>buttonDecrypt</tabstop>
  <tabstop>lineEditDNAhash</tabstop>
  <tabstop>scrollArea</tabstop>
  <tabstop>tabWidgetImage</tabstop>
  <tabstop>scrollArea_2</tabstop>
 </tabstops>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
